realm/src/main/java/io/realm/android/RealmCursor.java

```diff
@@ -0,0 +1,757 @@
+            return false;
+        }
+
+        // Verify position is not before the start of the cursor.
+        if (position < 0) {
+            rowIndex = -1;
+            return false;
+        }
+
+        rowIndex = position;
+        return true;
+    }
+
+    /**
+     * Move the cursor to the first row.
+     *
+     * <p>This method will return false if the cursor is empty.
+     *
+     * @return whether the move succeeded.
+     */
+    @Override
+    public boolean moveToFirst() {
+        return moveToPosition(0);
+    }
+
+    /**
+     * Move the cursor to the last row.
+     *
+     * <p>This method will return false if the cursor is empty.
+     *
+     * @return whether the move succeeded.
+     */
+    @Override
+    public boolean moveToLast() {
+        return moveToPosition(getCount() - 1);
+    }
+
+    /**
+     * Move the cursor to the next row.
+     *
+     * <p>This method will return false if the cursor is already past the
+     * last entry in the result set.
+     *
+     * @return whether the move succeeded.
+     */
+    @Override
+    public boolean moveToNext() {
+        return moveToPosition(rowIndex + 1);
+    }
+
+    /**
+     * Move the cursor to the previous row.
+     *
+     * <p>This method will return false if the cursor is already before the
+     * first entry in the result set.
+     *
+     * @return whether the move succeeded.
+     */
+    @Override
+    public boolean moveToPrevious() {
+        return moveToPosition(rowIndex - 1);
+    }
+
+    /**
+     * Returns whether the cursor is pointing to the first row.
+     *
+     * @return whether the cursor is pointing at the first entry.
+     */
+    @Override
+    public boolean isFirst() {
+        checkClosed();
+        return rowIndex == 0 && getCount() != 0;
+    }
+
+    /**
+     * Returns whether the cursor is pointing to the last row.
+     *
+     * @return whether the cursor is pointing at the last entry.
+     */
+    @Override
+    public boolean isLast() {
+        checkClosed();
+        int count = getCount();
+        return rowIndex == (count - 1) && count != 0;
+    }
+
+    /**
+     * Returns whether the cursor is pointing to the position before the first
+     * row.
+     *
+     * @return whether the cursor is before the first result.
+     */
+    @Override
+    public boolean isBeforeFirst() {
+        checkClosed();
+        if (getCount() == 0) {
+            return true;
+        }
+        return rowIndex == -1;
+    }
+
+    /**
+     * Returns whether the cursor is pointing to the position after the last
+     * row.
+     *
+     * @return whether the cursor is after the last result.
+     */
+    @Override
+    public boolean isAfterLast() {
+        checkClosed();
+        if (getCount() == 0) {
+            return true;
+        }
+        return rowIndex == getCount();
+    }
+
+    /**
+     * Returns the zero-based index for the given column name, or -1 if the column doesn't exist.
```

Comment:
I understand that SQLite will talk about columns, but Realm model classes have fields and not columns.